src/sync.rs

```diff
@@ -22,11 +22,14 @@ use std::{
     sync::{Once, OnceState},
 };
 
+pub mod critical_section;
 pub(crate) mod once_lock;
 
 #[cfg(not(Py_GIL_DISABLED))]
 use crate::PyVisit;
 
+#[deprecated(since = "0.28.0", note = "use std::sync::critical_section instead")]
+pub use self::critical_section::{with_critical_section, with_critical_section2};
```

Comment:
Sadly https://github.com/rust-lang/rust/issues/30827

One option is to define real functions here which just call the new ones; that deprecation will work properly.